internal/services/recoveryservices/backup_policy_file_share_resource.go

```diff
@@ -236,6 +249,52 @@ func resourceBackupProtectionPolicyFileShareRead(d *pluginsdk.ResourceData, meta
 					d.Set("retention_yearly", nil)
 				}
 			}
+
+			if properties.VaultRetentionPolicy != nil {
+				d.Set("backup_tier", "vault-standard")
+				if s := properties.VaultRetentionPolicy.SnapshotRetentionInDays; s != 0 {
+					d.Set("snapshot_retention_in_days", int(s))
+				} else {
+					d.Set("snapshot_retention_in_days", 0)
+				}
+
+				if retention, ok := properties.VaultRetentionPolicy.VaultRetention.(protectionpolicies.LongTermRetentionPolicy); ok {
+					if s := retention.DailySchedule; s != nil {
+						if err := d.Set("retention_daily", flattenBackupProtectionPolicyFileShareRetentionDaily(s)); err != nil {
+							return fmt.Errorf("setting `retention_daily`: %+v", err)
+						}
+					} else {
+						d.Set("retention_daily", nil)
+					}
```

Comment:
I appreciate this is how the existing retention_policy is flattened in this resource, but it would be good to fix this up to be consistent with how we handle flatten functions generally in the provider. Could we refactor all the flatten functions to do the nil check and return the interface if it is nil?